### ISSUE TITLE
Restructure project to be built with Bazel

### ISSUE THREAD

**Issue by rebeccajae** [rebeccajae/ethexplorer#10]
Currently this all lives at top-level, which is fine but it severely limits expandability for other adjacent projects that will eventually be merged into the ethexplorer ecosystem.

New projects either have to integrate tightly with this codebase, or create other repositories which makes things like breaking changes difficult to reason about.

Moving to use Bazel will provide a straightforward build system that supports other projects that will share code and direction with the main webapp. This will provide flexibility in direction as well as independence from a purely JS/TS codebase.

There are some upcoming plans to start developing other projects than ethexplorer to better deliver on the goals of a complete platform. I will begin filing those improvement tasks once this task is complete.
